tests/bsim/bluetooth/audio/src/cap_commander_test.c

```diff
@@ -195,6 +244,398 @@ static void cap_disconnected_cb(struct bt_conn *conn, uint8_t reason)
+	if (base == NULL) {
+		return true;
+	}
+
+	/* Can not fit all the received subgroups with the size CONFIG_BT_BAP_BASS_MAX_SUBGROUPS */
+	base_subgroup_count = bt_bap_base_get_subgroup_count(base);
+	if (base_subgroup_count < 0 || base_subgroup_count > CONFIG_BT_BAP_BASS_MAX_SUBGROUPS) {
+		printk("Got invalid subgroup count: %d\n", base_subgroup_count);
+		return true;
+	}
+
+	base_size = data->data_len - BT_UUID_SIZE_16; /* the BASE comes after the UUID */
+
+	/* Compare BASE and copy if different */
+	if (base_size != received_base_size || memcmp(base, received_base, base_size) != 0) {
+		(void)memcpy(received_base, base, base_size);
+		received_base_size = base_size;
+	}
+
+	/* Stop parsing */
+	return false;
+}
+
+static void pa_recv(struct bt_le_per_adv_sync *sync,
+		    const struct bt_le_per_adv_sync_recv_info *info, struct net_buf_simple *buf)
+{
+	if (TEST_FLAG(flag_base_received)) {
+		return;
+	}
+
+	bt_data_parse(buf, base_store, NULL);
+	SET_FLAG(flag_base_received);
+}
+
+static struct bt_le_per_adv_sync_cb bap_pa_sync_cb = {
+	.synced = bap_pa_sync_synced_cb,
+	.term = bap_pa_sync_terminated_cb,
+	.recv = pa_recv,
+};
+
+static void base_recv_cb(struct bt_bap_broadcast_sink *sink, const struct bt_bap_base *base,
+			 size_t base_size)
+{
+	uint32_t base_bis_index_bitfield = 0U;
+	int ret;
+
+	if (TEST_FLAG(flag_base_received)) {
+		return;
+	}
+
+	ret = bt_bap_base_get_subgroup_count(base);
+	if (ret < 0) {
+		FAIL("Failed to get subgroup count: %d\n", ret);
+		return;
+	}
+
+	if (ret == 0) {
+		FAIL("subgroup_count was 0");
+		return;
+	}
+
+	ret = bt_bap_base_foreach_subgroup(base, valid_subgroup_metadata_cb, NULL);
+	if (ret != 0) {
+		FAIL("Failed to parse subgroups: %d\n", ret);
+		return;
+	}
+
+	ret = bt_bap_base_get_bis_indexes(base, &base_bis_index_bitfield);
+	if (ret != 0) {
+		FAIL("Failed to BIS indexes: %d\n", ret);
+		return;
+	}
+
+	bis_index_bitfield = base_bis_index_bitfield & bis_index_mask;
+
+	SET_FLAG(flag_base_received);
+}
+
+static void syncable_cb(struct bt_bap_broadcast_sink *sink, const struct bt_iso_biginfo *biginfo)
+{
+	printk("Broadcast sink %p syncable with%s encryption\n", sink,
+	       biginfo->encryption ? "" : "out");
+	SET_FLAG(flag_syncable);
+}
+
+static struct bt_bap_broadcast_sink_cb broadcast_sink_cbs = {
+	.base_recv = base_recv_cb,
+	.syncable = syncable_cb,
+};
+
+static void bap_broadcast_assistant_discover_cb(struct bt_conn *conn, int err,
+						uint8_t recv_state_count)
+{
+	if (err == 0) {
+		printk("BASS discover done with %u recv states\n", recv_state_count);
+	} else {
+		printk("BASS discover failed (%d)\n", err);
+	}
+	k_sem_give(&sem_ba_discovered);
+}
+
+static void bap_broadcast_assistant_add_src_cb(struct bt_conn *conn, int err)
+{
+	if (err == 0) {
+		printk("BASS add source successful\n");
+	} else {
+		printk("BASS add source failed (%d)\n", err);
+	}
+}
+
+static bool metadata_entry(struct bt_data *data, void *user_data)
+{
+	char metadata[512];
```

Comment:
```suggestion
	char metadata[UINT8_MAX];
```
Metadata can maximum be 254 octets

Reply:
We should probably actually use `CONIFG_BT_AUDIO_CODEC_CFG_MAX_METADATA_SIZE` instead, as we do not support writing more than that, right?

Reply:
Is this function even necessary? Do we care about the metadata?

Reply:
This was copied from the broadcast_assistant_test

So, we should discuss whether we want to refactor and create some common callback functions, or if we want to finetune the callbacks for each test

Reply:
> This was copied from the broadcast_assistant_test, so that one may then also need an update.

Not sure if I agree. The BAP broadcast assistant tests BAP which is where the metadata is handled. It makes more sense to verify the behavior there (although it should have been an actual test rather than a print), than at the CAP level.